packages/zeebe-element-templates-json-schema/src/defs/template.json

```diff
@@ -260,6 +260,209 @@
+                            "required": ["type", "property"]
+                          }
+                        },
+                        "required": ["binding"]
+                      }
+                    }
+                  }
+                },
+                {
+                  "properties": {
+                    "properties": {
+                      "contains": {
+                        "properties": {
+                          "binding": {
+                            "properties": {
+                              "type": { "const": "zeebe:taskDefinition" },
+                              "property": { "enum": ["type", "retries"] }
+                            },
+                            "required": ["type", "property"]
+                          }
+                        },
+                        "required": ["binding"]
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+          },
+          {
+            "if": {
+              "properties": {
+                "properties": {
+                  "contains": {
+                    "properties": {
+                      "binding": {
+                        "properties": {
+                          "type": { "const": "zeebe:taskDefinition" }
+                        },
+                        "required": ["type"]
+                      }
+                    },
+                    "required": ["binding"]
+                  }
+                }
+              }
+            },
+            "then": {
+              "properties": {
+                "properties": {
+                  "allOf": [
+                    {
+                      "contains": {
+                        "properties": {
+                          "binding": {
+                            "properties": {
+                              "type": { "const": "zeebe:taskDefinition" },
+                              "property": { "const": "type" }
+                            },
+                            "required": ["type", "property"]
+                          }
+                        },
+                        "required": ["binding"]
+                      }
+                    },
+                    {
+                      "contains": {
+                        "properties": {
+                          "binding": {
+                            "properties": {
+                              "type": { "const": "zeebe:taskDefinition" },
+                              "property": { "const": "retries" }
+                            },
+                            "required": ["type", "property"]
+                          }
+                        },
+                        "required": ["binding"]
+                      }
+                    }
+                  ]
+                }
+              }
+            }
+          }
+        ],
+        "required": [
+          "properties"
+        ]
+      }
```

Comment:
this should allow using both 'zeebe:script' (expression, result) and 'zeebe:taskDefinition' (type, retries) - but I'm not sure I've used the proper syntax as tests are only partially working as expected